hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DefaultSerializationServiceBuilder.java

```diff
@@ -125,6 +130,9 @@ public SerializationServiceBuilder setConfig(SerializationConfig config) {
         enableCompression = config.isEnableCompression();
         enableSharedObject = config.isEnableSharedObject();
         allowUnsafe = config.isAllowUnsafe();
+        JavaSerializationFilterConfig filterConfig = config.getJavaSerializationFilterConfig();
+        classNameFilter = filterConfig == null ? SerializationClassNameFilter.DEFAULT_CLASSNAME_FILTER
```

Comment:
This defaulting logic should be moved in the config. And in this case since always a filter is created, the config should not allow for a null javaSerializationFilterConfig to be returned.